Server/src/main/java/org/openas2/cmd/processor/restapi/ApiResource.java

```diff
@@ -220,6 +247,139 @@ public Response headCommand(@PathParam("param") String command) {
+
+
+            XPath xpath = XPathFactory.newInstance().newXPath();
+            XPathExpression expr = xpath.compile(xpathExpression);
+            NodeList nodeList = (NodeList) expr.evaluate(document, XPathConstants.NODESET);
+
+
+            Document resultDocument = builder.newDocument();
+            Element root = resultDocument.createElement("results");
+            resultDocument.appendChild(root);
+
+            for (int i = 0; i < nodeList.getLength(); i++) {
+                Node importedNode = resultDocument.importNode(nodeList.item(i), true);
+                root.appendChild(importedNode);
+            }
+
+            // Convert to string
+            Transformer transformer = TransformerFactory.newInstance().newTransformer();
+            StringWriter stringWriter = new StringWriter();
+            transformer.transform(new DOMSource(resultDocument), new StreamResult(stringWriter));
+
+            return Response.ok(stringWriter.toString(), MediaType.APPLICATION_XML).build();
+
+        } catch (Exception exception) {
+            LoggerFactory.getLogger(ApiResource.class.getName()).error(
+                    "Error processing XML file: " + filePath, exception);
+            return Response.serverError()
+                    .entity("{\"error\":\"Internal server error\"}")
+                    .type(MediaType.APPLICATION_JSON)
+                    .build();
+        }
+    }
+
+
+
+    private static boolean isLocalhost(Request request) {
+        boolean isLocalhost = request.getRemoteAddr().equals("127.0.0.1") || request.getRemoteAddr().equals("::1");
+        return isLocalhost;
+    }
+
+    private NodeList getNodes(String xmlFileName, String xpathExpression) {
+        NodeList nodeList = null;
+        try {
+            DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+
+            // === XXE Protection ===
+            dbf.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
+            dbf.setFeature("http://xml.org/sax/features/external-general-entities", false);
+            dbf.setFeature("http://xml.org/sax/features/external-parameter-entities", false);
+            dbf.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
+            dbf.setXIncludeAware(false);
+            dbf.setExpandEntityReferences(false);
+
+            DocumentBuilder db = dbf.newDocumentBuilder();
+            File file = new File(xmlFileName);
+            Document document = db.parse(file);
+
+            XPathExpression xPathExpr = XPathFactory.newInstance().newXPath().compile(xpathExpression);
+            nodeList = (NodeList) xPathExpr.evaluate(document, XPathConstants.NODESET);
+
+        } catch (Exception ex) {
+            LoggerFactory.getLogger(ApiResource.class.getName()).error("Error parsing XML file: " + xmlFileName, ex);
+            // return null on error
+        }
+        return nodeList;
+    }
```

Comment:
This appears to be a redundant method.
What is the reason this is here?

Reply:
This method is not called from anywhere so I am unclear why it is here unless you have other code that is not included in this PR.